internal/olm/client.go

```diff
@@ -0,0 +1,467 @@
+}
+
+func (c Client) GetStatus(ctx context.Context, version string) (*Status, error) {
+	resources, err := c.getResources(ctx, version)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get resources: %s", err)
+	}
+
+	status := c.getStatus(ctx, resources)
+	if !status.HasExistingResources() {
+		return nil, fmt.Errorf("no existing installation found")
+	}
+	return &status, nil
+}
+
+func (c Client) getResources(ctx context.Context, version string) ([]unstructured.Unstructured, error) {
+	log.Infof("Fetching CRDs for version %s", version)
+	crdResources, err := c.getCRDs(ctx, version)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch CRDs: %s", err)
+	}
+
+	log.Infof("Fetching resources for version %s", version)
+	olmResources, err := c.getOLM(ctx, version)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch resources: %s", err)
+	}
+
+	resources := append(crdResources, olmResources...)
+	return resources, nil
+}
+
+func (c Client) getStatus(ctx context.Context, objs []unstructured.Unstructured) Status {
+	var rss []ResourceStatus
+	for _, obj := range objs {
+		nn := types.NamespacedName{
+			Namespace: obj.GetNamespace(),
+			Name:      obj.GetName(),
+		}
+		u := unstructured.Unstructured{}
+		u.SetGroupVersionKind(obj.GroupVersionKind())
+		err := c.KubeClient.Get(ctx, nn, &u)
+		rs := ResourceStatus{
+			NamespacedName: nn,
+			GVK:            obj.GroupVersionKind(),
+		}
+		if err != nil {
+			rs.Error = err
+		} else {
+			rs.Resource = &u
+		}
+		rss = append(rss, rs)
+	}
+
+	return Status{Resources: rss}
+}
+
+func (c Client) resolveVersion(ctx context.Context, version string) (string, error) {
+	var url string
+	if version == "latest" {
+		url = fmt.Sprintf("%s/%s", c.BaseReleasesAPIURL, version)
+	} else {
+		url = fmt.Sprintf("%s/tags/%s", c.BaseReleasesAPIURL, version)
+	}
+
+	resp, err := c.doRequest(ctx, url)
+	if err != nil {
+		return version, fmt.Errorf("request failed: %s", err)
+	}
+	defer resp.Body.Close()
+
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return version, fmt.Errorf("read response body: %s", err)
+	}
+
+	var d struct {
+		TagName string `json:"tag_name"`
+	}
+	if err := json.Unmarshal([]byte(data), &d); err != nil {
+		return version, fmt.Errorf("unmarshal json: %s", err)
+	}
+	return d.TagName, nil
+}
+
+func (c Client) getCRDs(ctx context.Context, version string) ([]unstructured.Unstructured, error) {
+	resp, err := c.doRequest(ctx, c.crdsURL(version))
+	if err != nil {
+		return nil, fmt.Errorf("request failed: %s", err)
+	}
+	defer resp.Body.Close()
+	return decodeResources(resp.Body)
+}
+
+func (c Client) getOLM(ctx context.Context, version string) ([]unstructured.Unstructured, error) {
+	resp, err := c.doRequest(ctx, c.olmURL(version))
+	if err != nil {
+		return nil, fmt.Errorf("request failed: %s", err)
+	}
+	defer resp.Body.Close()
+	return decodeResources(resp.Body)
+}
+
+func (c Client) doRequest(ctx context.Context, url string) (*http.Response, error) {
```

Comment:
I would just suggest adding an e2e test in case a breaking change happens to the URLs or the location of the manifest files. WDYT? It's not a blocker from my side, just a suggestion.

Reply:
ha ignore above, I missed the test file :) 👍